sql/core/src/test/scala/org/apache/spark/sql/execution/BaseScriptTransformationSuite.scala

```diff
@@ -670,6 +670,26 @@ abstract class BaseScriptTransformationSuite extends SparkPlanTest with SQLTestU
       ),
       df.select($"col").collect())
   }
+
+  test("SPARK-41790: Set TRANSFORM reader and writer's format correctly") {
+    withTempView("v") {
+      val df = Seq(
+        (1, 2)
+      ).toDF("a", "b")
+      df.createTempView("v")
+
+      checkAnswer(
+        sql(
+          s"""
+             |SELECT TRANSFORM(a, b)
+             |  ROW FORMAT DELIMITED
+             |  FIELDS TERMINATED BY ','
+             |  USING 'cat'
+             |  AS (c)
+             |FROM v
+        """.stripMargin), identity, Row("1,2") :: Nil)
```

Comment:
Reader, writer seems only used in HiveScriptTransformationExec, this UT should move to `HiveScriptTransformationSuite`

Reply:
Can you explain why this UT can verify the change?

Reply:
This UT will get nothing in currently master branch code, as [SPARK-41790](https://issues.apache.org/jira/projects/SPARK/issues/SPARK-41790) description said.

Reply:
> This UT will get nothing in currently master branch code, as [SPARK-41790](https://issues.apache.org/jira/projects/SPARK/issues/SPARK-41790) description said.

Can you add desc in jira into this pr's desc?

Reply:
ok